### PR TITLE
Make HTTPIOFile an subclass of io.BufferedIOBase

### PR DESCRIPTION
**Pull request by jamesba**

### CODE REVIEW

**Comment by barneygale** [comment]
I've merged #1, does this now need a rebase?

**Comment by jamesba** [comment]
Rebase done

File: httpio.py [hunk 1]
```diff
@@ -1,6 +1,19 @@
+from __future__ import print_function
```
Comment (barneygale) [inline httpio.py]: Unused import?

File: httpio.py [hunk 2]
```diff
@@ -1,6 +1,19 @@
...
+from __future__ import absolute_import
+
 import re
 import requests
 
+from io import BufferedIOBase
+
+from six import PY3
```
Comment (barneygale) [inline httpio.py]: Do we need to add six as a dependency in setup.py?

File: httpio.py [hunk 3]
```diff
@@ -72,35 +86,77 @@ def read(self, size=-1):
...
-
-            data = b"".join(data)
-
-        self._cursor += size
+            data = b''.join(self._read_cached(size))
+
+        self._cursor += len(data)
         return data
 
+    def read1(self, size=-1):
```
Comment (barneygale) [inline httpio.py]: Quite a lot of code duplication here and in `readinto1` - perhaps it's worth refactoring `read()` and `read1()` to be based on some `_read_impl()`, or something like that? If that's too big an ask I can do it as a post-commit thing.

**Comment by barneygale** [comment]
Thanks very much, and sorry again for the slow progress.

**Pull request merged**